### PR TITLE
Refactor test helpers

### PR DESCRIPTION
**Pull request by khardix**
An implementation of sclorg#57.

General description of changes:
- Replaced custom setup/teardown logic with fixtures (built-in or custom) where possible
- Moved global helper functions to dedicated, auto-discovered namespace
- Replaced `str` with `pathlib.Path` for file-system path manipulation where possible

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -88,3 +88,27 @@ def generate_tmp_path_list(count: int):
...
+
+@pytest.fixture
+def random_file_path():
+    """Provide path to a random temporary file."""
+
+    with generate_tmp_path() as path:
+        yield path
+
+
+@pytest.fixture
```
Comment (hroncok) [inline tests/conftest.py]: Could this be session scoped? Paths are immutable.

File: tests/conftest.py [hunk 2]
```diff
@@ -88,3 +88,27 @@ def generate_tmp_path_list(count: int):
...
+def valid_recipe_path():
+    """Provide path to a valid recipe file."""
+
+    return Path('tests/fixtures/recipes/ror.yml')
+
+
+@pytest.fixture
```
Comment (hroncok) [inline tests/conftest.py]: Here as well.

Comment (khardix) [inline tests/conftest.py]: It can be, in both places.

File: tests/test_recipe.py [hunk 3]
```diff
@@ -3,13 +3,12 @@
...
 
-import helper
 from rpmlb.recipe import Recipe
 
 
 @pytest.fixture
-def ok_recipe():
-    return Recipe(helper.get_valid_recipe_file(), 'rh-ror50')
+def ok_recipe(valid_recipe_path):
+    return Recipe(str(valid_recipe_path), 'rh-ror50')
```
Comment (hroncok) [inline tests/test_recipe.py]: Note to self: Would be good if Recipe could take a PathLib object.

Comment (khardix) [inline tests/test_recipe.py]: It would, but since the `os.*` functions do not take PathLib object too (on Python 3.4), it is low priority issue…

**Pull request merged**